0003-eliminate-stats-record.md

```diff
@@ -0,0 +1,34 @@
+# Eliminate stats.Record functionality
+
+Remove `stats.Record` from the specification, following the MeasureMetric type (RFC 0002-metric-measure).
+
+## Motivation
+
+This change is no longer a necessary interface. There are conceivable reasons to support it, but they are not outweighed by the cost of implementing and supporting two interfaces for recording metrics and statistics.
+
+## Explanation
+
+In RFC 0002-metric-measure, a new MeasureMetric type is introduced to replace raw statistics, with support for pre-defined label values.  With the new type introduced, it's now possible to record formerly-raw statistics through a higher-level Metric interface.  stats.Record is no longer a requirement and should be removed until strong evidence of its need emerges from real use-cases.
+
+## Internal details
+
+This simply involves removing the low-level `stats.Record` API from the specification.
+
+## Trade-offs and mitigations
+
+There are two reasons to maintain a low-level API that we know of:
+
+1. For _generality_.  An application that forwards metrics from another source may need to handle metrics in generic code.  For these applications, having type-specific Metric handles could actually require more code to be written, whereas the low-level `stats.Record` API is more amenable to generic use.
+1. For _atomicity_.  An application that wishes to record multiple statistics in a single operation can feel confident computing formulas based on multiple metrics, not worry about inconsistent views of the data.
+
+## Prior art and alternatives
+
+Raw statistics were a solution to confusion found in existing metrics APIs over Metric types vs. Aggregation types.  This proposal accompanies RFC 0001-metric-pre-defined-labels and RFC 0002-metric-measure.md in proposing that we think about Metric _type_ as independent of which aggregations apply.  Once we have a Metric to support histogram and summary aggregations, we no longer need raw statistics, and we no longer need `stats.Record`.  This avoids introducing new concepts (Raw statistics), at the same time departs from prior art in letting one Metric type support both Histogram and Summary aggregations.
+
+## Open questions
+
+Are either of the trade-offs described above important enough to keep the low-level `stats.Record` API?
+
+## Future possibilities
+
+This restricts future possibilities for the benefit of a smaller, simpler specification.
```

Comment:
Arguably, we can always reduce `stats.Record` (or something with equivalent functionality) back later if there's user need. Adding it back should be a lot easier than deprecating it 18 months from now would be :)

Reply:
(I moved that statement here, from elsewhere.)